lib/rules/max-length-import-relative-path.js

```diff
@@ -0,0 +1,65 @@
+/**
+ * @fileoverview Import local JS modules without `../'ing`.
+ * @author Alan Accurso
+ */
+"use strict";
+
+//------------------------------------------------------------------------------
+// Rule Definition
+//------------------------------------------------------------------------------
+
+module.exports = {
+    meta: {
+        docs: {
+            description: "Import local JS modules without `../'ing`.",
+            category: "Module System",
+            recommended: false
+        },
+        fixable: null,  // or "code" or "whitespace"
+        schema: [
+            {
+                type: 'object',
+                properties: {
+                    max: {
+                        type: 'number'
+                    }
+                },
+                additionalProperties: false
+            }
+        ]
+    },
+
+    errors: {
+        MAX_NUMBER_OF_PARENT_DIRECTORIES_EXCEEDED: 'Max number of parent directories exceeded. Expected less than {{max}} but found {{numberOfParentDirectories}}.'
+    },
+
+    create: function(context) {
+
+        const that = this;
+        const { max = 0 } = context.options[0] || {};
+        const PARENT_DIRECTORY = '../';
+
+        //----------------------------------------------------------------------
+        // Public
+        //----------------------------------------------------------------------
+
+        return {
+
+            ImportDeclaration: node => {
+                const numberOfParentDirectories = node.source.value.split(PARENT_DIRECTORY).length - 1;
+
+                if (numberOfParentDirectories > max) {
+                    context.report(
+                        node.source,
+                        that.errors.MAX_NUMBER_OF_PARENT_DIRECTORIES_EXCEEDED,
```

Comment:
So this can just be... `this`.

Reply:
Actually, `this` in the scope of `ImportDeclaration` is the returned object, not the module itself. I'll extract the errors object outside the module so it's in the scope of all functions. Let me know what you think about it.

Reply:
This is a fine solution, but the arrow function binds to the lexical scope, so it inherits `this` from where it's declared. So in your previous code, it would not be bound to the returned object.

Reply:
You are right, then `eslint` must be binding the returned object to something else, because I can't access the `errors` property with `this.errors`.

So this seems to be an `eslint` restriction.